src/main/java/com/denizenscript/clientizen/scripts/containers/gui/elements/SliderElement.java

```diff
@@ -0,0 +1,108 @@
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getTaggedInt;
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getWidgetId;
+
+public class SliderElement implements GuiScriptContainer.GuiElementParser {
+
+    // <--[language]
+    // @name Slider GUI Element
+    // @group GUI System
+    // @description
+    // Sliders allow controlling a number (integer) value, with a customizable min/max values; they have a UI type of "slider".
+    //
+    // <code>
+    // ui_type: slider
+    // # The minimum value the slider can hold, optional (defaults to 0).
+    // min: <number>
+    // # The maximum value the slider can hold, required.
+    // max: <number>
+    // # The axis the slider slides along, required.
+    // axis: HORIZONTAL/VERTICAL
+    // # The direction the slider slides in along its axis to increase its value, optional.
+    // # Defaults to UP for vertical sliders, and RIGHT for horizontal sliders.
+    // direction: UP/DOWN/LEFT/RIGHT
+    // # The value the slider should be on, optional.
+    // value: <number>
+    // # Code to run when the slider's value is changed in the slightest (each number it goes through while dragging), optional.
+    // # Provides <context.new_value>, returning an ElementTag(Number) of the slider's new value.
+    // # Should generally prefer listening to the final value set instead of this.
+    // on_change:
+    // - <script>
+    // # Code to run when the slider's value is set (the user finished changing it), optional.
+    // # Provides <context.new_value>, returning an ElementTag(Number) of the slider's new value.
+    // on_set:
+    // - <script>
+    // </code>
+    // -->
+
+    @Override
+    public WWidget parse(GuiScriptContainer container, YamlConfiguration config, String pathToElement, TagContext context) {
+        return parseSlider(WSlider::new, container, config, pathToElement, context);
+    }
+
+    public static <T extends WAbstractSlider> T parseSlider(TriFunction<Integer, Integer, Axis, T> constructor, GuiScriptContainer container, YamlConfiguration config, String pathToElement, TagContext context) {
+        Integer min = getTaggedInt(config, "min", 0, context);
+        if (min == null) {
+            return null;
+        }
+        Integer max = getTaggedInt(config, "max", context);
+        if (max == null) {
+            Debug.echoError("Must specify a max value");
+            return null;
+        }
+        Axis axis = getTaggedEnum(Axis.class, config, "axis", context);
+        if (axis == null) {
+            Debug.echoError("Must specify an axis.");
+            return null;
+        }
+        T slider = constructor.apply(min, max, axis);
+        WAbstractSlider.Direction direction = getTaggedEnum(WAbstractSlider.Direction.class, config, "direction", context);
+        if (direction != null) {
+            slider.setDirection(direction);
+        }
```

Comment:
what happens when uo/down used for horizontal, or left/right used for vertical?

Reply:
Added proper meta & errors for this